### PR TITLE
Votes API test

### LINKED ISSUE #39: Votes: Create Test coverage

### PR DESCRIPTION
**Pull request by fmSand**
Test coverage for:

- Adding/removing upvotes on a project
- Multiple users voting on the same project
- Individual vote toggling (one user’s vote does not affect others)
- Repeated toggle operations (on/off/on)
- Authentication and validation
- Error handling for unauthenticated and non-existent project requests

**10/10 tests pass**

Waiting for team decision on allowing for self-upvote or not. If No - I'll implement this and add a test for it later :)

### CODE REVIEW

**Comment by Anclagen** [comment]
When testing middleware, the goal isn’t necessarily to test the middleware itself, but to confirm that it’s correctly applied to the route. You're verifying that users can’t send bad requests, perform unauthorized actions, or modify resources they don't own.

I recommend including at least one negative test per middleware type (auth, validation, ownership) on each route. This ensures the middleware is actually in place and functioning at a basic level.

As a stretch goal, we can add more targeted middleware tests, especially for more complex logic so we can ensure they handle edge cases and behave consistently.

File: tests/votes.test.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+const createTestUser = (suffix = "") => ({
+  id: `33333333-3333-3333-3333-33333333333${suffix || "3"}`,
+  firstName: "Vote",
+  lastName: `Tester${suffix}`,
+  username: `votetester${suffix}`,
+  email: `votetester${suffix}@example.com`,
+  hashedPassword: "testhash",
+  salt: "testsalt",
+});
```
Comment (Anclagen) [inline tests/votes.test.js:14]: This is a nice way to generate bulk users, will steal this for my project tests.

File: tests/votes.test.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  it("allows multiple users to upvote same project", async () => {
+    const res1 = await toggleUpvote(projectId, token1);
+    expect(res1.body.data.upvoted).toBe(true);
+    const countAfterUser1 = res1.body.data.count;
+
+    const res2 = await toggleUpvote(projectId, token2);
+    expect(res2.body.data.upvoted).toBe(true);
+    expect(res2.body.data.count).toBe(countAfterUser1 + 1);
+  });
```
Comment (Anclagen) [inline tests/votes.test.js:88]: Should we be allowing the project creator to upvote their own projects?

File: tests/votes.test.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  it("rejects vote request for a non-existent project", async () => {
+    const nonExistentProjectId = "99999999-9999-9999-9999-999999999999";
+    const res = await toggleUpvote(nonExistentProjectId, token1);
+
+    expect(res.statusCode).toBe(404);
+    expect(res.body.success).toBe(false);
+    expect(res.body.status).toBe("fail");
+  });
```
Comment (Anclagen) [inline tests/votes.test.js]: We might want to add validation to the project ID on the votes route so non-uuid are thrown out before checking the database? and then add an additional test like this.

File: tests/votes.test.js (hunk 2 above)
Comment (fmSand) [reply tests/votes.test.js:88]: I'm fine with either or. Will wait with implementing it till team decides.

File: tests/votes.test.js (hunk 3 above)
Comment (fmSand) [reply tests/votes.test.js]: Done

**Review by Anclagen — APPROVED** [review]
This all looks good for the moment.
